googletakeout.cc

```diff
@@ -0,0 +1,416 @@
+#include <QDir>
+#include <QIODevice>
+#include <QJsonObject>
+#include <QJsonArray>
+#include <QJsonParseError>
+
+#include "src/core/file.h"
+
+#include "googletakeout.h"
+
+#define MYNAME "Google Takeout"
+#define TIMELINE_OBJECTS "timelineObjects"
+#define PLACE_VISIT "placeVisit"
+#define ACTIVITY_SEGMENT "activitySegment"
+#define ACTIVITY_TYPE "activityType"
+#define LOCATION "location"
+#define LOCATION_LATE7 "latitudeE7"
+#define LOCATION_LONE7 "longitudeE7"
+#define NAME "name"
+#define ADDRESS "address"
+#define DURATION "duration"
+#define START_TIMESTAMP "startTimestamp"
+#define START_LOCATION "startLocation"
+#define END_TIMESTAMP "endTimestamp"
+#define END_LOCATION "endLocation"
+#define TIMESTAMP "timestamp"
+#define SIMPLE_PATH "simplifiedRawPath"
+#define POINTS "points"
+#define WAYPOINT_PATH "waypointPath"
+#define WAYPOINTS  "waypoints"
+// for some reason that probably only a former Google engineer knows,
+// we use "latE7"/"lngE7" here instead of "latitudeE7"/"longitudeE7".
+// +10 points for brevity, but -100 points for inconsistency.
+#define LATE7 "latE7"
+#define LONE7 "lngE7"
```

Comment:
can we make these private static class constants?
 used to declare a constant; consider using a 'constexpr' constant [cppcoreguidelines-macro-usage]